File: malipo-php/src/Resources/Balance.php
<?php

namespace Malipo\Resources;

use Malipo\Malipo;

class Balance
{
    private $client;

    public function __construct(Malipo $client)
    {
        $this->client = $client;
    }

    public function retrieve()
    {
        $endpoint = $this->client->getEnvironment() === 'live' ? '/live-balance' : '/sandbox-balance';
        return $this->client->request('GET', $endpoint);
    }
}

File: malipo-flutter/lib/src/resources/balance.dart
import '../client.dart';
import '../models.dart';

class BalanceResource {
  final Malipo client;

  BalanceResource(this.client);

  Future<MalipoBalance> retrieve() async {
    final endpoint = client.environment == 'live' ? '/live-balance' : '/sandbox-balance';
    final response = await client.request('GET', endpoint);

    return MalipoBalance.fromJson(response);
  }
}
